source/common/network/lc_trie.h

```diff
@@ -226,16 +225,12 @@ class LcTrie {
       std::sort(tag_data.begin(), tag_data.end());
       ip_prefixes_.push_back(tag_data[0]);
 
-      // Remove duplicate entries and check for nested prefixes.
+      // Set up ip_prefixes_, which should contain the supplied prefixes in sorted order,
+      // but with any nested prefixes encapsulated under their parent prefixes.
       for (size_t i = 1; i < tag_data.size(); ++i) {
-        // TODO(ccaraman): Add support for nested prefixes.
-        if (tag_data[i - 1].isPrefix(tag_data[i])) {
-          throw EnvoyException(fmt::format(
-              "LcTrie does not support nested prefixes. '{0}' is a nested prefix of '{1}'.",
-              tag_data[i].asString(), tag_data[i - 1].asString()));
-        }
-
-        if (tag_data[i - 1] != tag_data[i]) {
+        if (ip_prefixes_[ip_prefixes_.size() - 1].isPrefix(tag_data[i])) {
+          ip_prefixes_[ip_prefixes_.size() - 1].addNestedPrefix(tag_data[i]);
```

Comment:
Ok, question time, if we're trying to build a trie, isn't this actually building a vector of maybe vectors?  I'd expect from the class name if we had nested prefixes those could have their own nested prefix rather than being bundled together at depth 1.   Is this just infrequent enough we do it the inefficient way and if so, mind commenting it up?

Reply:
Yes, it is building a vector of nested prefixes. From talking to @mattklein123, nested prefixes aren't that common and linearly traversing a sorted vector in these cases should be good enough for now. If we do notice there being an issue, we can always swap this to have tries for the nested prefix cases.

Reply:
I'll add a `TODO` comment about how more general-purpose usage will require using sub-tries for the nested prefix cases.

Reply:
Another solution, I think, would be to change the construction of the base LC Trie to allow nested prefixes. As far as I can tell, the data structure should be able to handle nested prefixes in a single trie; it's just the trie construction algorithm from the original paper that doesn't support nesting. For now, though, I just added the `TODO` comment.